drivers/SmartThings/zwave-garage-door-opener/src/ecolink-zw-gdo/init.lua

```diff
@@ -0,0 +1,309 @@
+---
+--- @param driver st.zwave.Driver
+--- @param device st.zwave.Device
+--- @param cmd st.zwave.CommandClass.Notification.Report
+local function notification_report_handler(driver, device, cmd)
+  local notificationType = cmd.args.notification_type
+  local notificationEvent = cmd.args.event
+  local barrier_event = nil
+  local contact_event = nil
+  if ( 0 == notificationEvent ) then
+    -- Clear Notifications
+    -- First byte of the parameters is the notification being cleared
+    -- so reuse notificationEvent variable as the event being cleared
+    if (0 ~= string.len(cmd.args.event_parameter)) then
+      notificationEvent = string.byte(cmd.args.event_parameter)
+    end
+    if (notificationType == Notification.notification_type.SYSTEM) then
+      if (notificationEvent == Notification.event.system.TAMPERING_PRODUCT_COVER_REMOVED) then
+        contact_event = capabilities.tamperAlert.tamper.clear()
+      end
+    elseif (notificationType == Notification.notification_type.ACCESS_CONTROL) then
+      if (notificationEvent ==
+          Notification.event.access_control.BARRIER_SENSOR_LOW_BATTERY_WARNING) then
+        barrier_event = capabilities.doorControl.door.closed()
+        contact_event = capabilities.battery.battery(CONTACTSENSOR_BATTERY_LEVEL_NORMAL)
+      elseif (notificationEvent ==
+              Notification.event.access_control.BARRIER_SENSOR_NOT_DETECTED_SUPERVISORY_ERROR) then
+        barrier_event = capabilities.doorControl.door.closed()
+      end
+    end
+  else
+    -- Handle Notification events
+    if (notificationType == Notification.notification_type.SYSTEM) then
+      if (notificationEvent == Notification.event.system.TAMPERING_PRODUCT_COVER_REMOVED) then
+        contact_event = capabilities.tamperAlert.tamper.detected()
+      end
+    elseif (notificationType == Notification.notification_type.ACCESS_CONTROL) then
+      if (notificationEvent == Notification.event.access_control.WINDOW_DOOR_IS_OPEN) then
+        barrier_event = capabilities.doorControl.door.open()
+        contact_event = capabilities.contactSensor.contact.open()
+      elseif (notificationEvent == Notification.event.access_control.WINDOW_DOOR_IS_CLOSED) then
+        barrier_event = capabilities.doorControl.door.closed()
+        contact_event = capabilities.contactSensor.contact.closed()
+      elseif (notificationEvent ==
+      Notification.event.access_control.BARRIER_MOTOR_HAS_EXCEEDED_MANUFACTURERS_OPERATIONAL_TIME_LIMIT) then
+        barrier_event = capabilities.doorControl.door.unknown()
+      elseif (notificationEvent ==
+      Notification.event.access_control.BARRIER_UNABLE_TO_PERFORM_REQUESTED_OPERATION_DUE_TO_UL_REQUIREMENTS) then
+        barrier_event = capabilities.doorControl.door.unknown()
+      elseif (notificationEvent ==
+      Notification.event.access_control.BARRIER_FAILED_TO_PERFORM_REQUESTED_OPERATION_DEVICE_MALFUNCTION) then
+        barrier_event = capabilities.doorControl.door.unknown()
```

Comment:
could these three if statements with the same result be combined?

Reply:
Yes they can, I made the change.